yogstation/code/modules/antagonists/darkspawn/darkspawn_abilities/warlock_abilities.dm

```diff
@@ -0,0 +1,549 @@
+	if(target.stat & BROKEN)
+		to_chat(owner, span_danger("This [target] no longer functions enough for access to the power grid."))
+		return FALSE
+	return TRUE	
+
+/datum/action/cooldown/spell/touch/null_charge/cast_on_hand_hit(obj/item/melee/touch_attack/hand, obj/machinery/power/apc/target, mob/living/carbon/human/caster)
+	if(!target || !istype(target))//sanity check
+		return FALSE
+
+	//Turn it off for the time being
+	to_chat(owner, span_velvet("Xlahwa..."))
+	target.set_light(0)
+	target.visible_message(span_warning("The [target] flickers and begins to grow dark."))
+
+	to_chat(caster, span_velvet("You dim the APC's screen and carefully begin siphoning its power into the void."))
+	if(!do_after(caster, 5 SECONDS, target))
+		//Whoops!  The APC's light turns back on
+		to_chat(caster, span_velvet("Your concentration breaks and the APC suddenly repowers!"))
+		target.set_light(2)
+		target.visible_message(span_warning("The [target] begins glowing brightly!"))
+		return FALSE
+
+	if(target.stat & BROKEN)
+		to_chat(owner, span_danger("This [target] no longer functions enough for access to the power grid."))
+		return FALSE
+
+	//We did it
+	if(isdarkspawn(owner))
+		var/datum/antagonist/darkspawn/darkspawn = isdarkspawn(owner)
+		darkspawn.block_psi(60 SECONDS, type)
+	to_chat(owner, span_progenitor("...SHWOOH!"))
+	priority_announce("Abnormal activity detected in [station_name()]'s powernet. As a precautionary measure, the station's power will be shut off for an indeterminate duration.", "Critical Power Failure", ANNOUNCER_POWEROFF)
+	power_fail(30, 40)
+	to_chat(caster, span_velvet("You return the APC's power to the void, destroying it and disabling all others."))
+	target.set_broken()
+	return TRUE
+		
+//////////////////////////////////////////////////////////////////////////
+//-----------------------Drain enemy, heal ally-------------------------//
+//////////////////////////////////////////////////////////////////////////
+/datum/action/cooldown/spell/pointed/extract
+	name = "Extract"
+	desc = "Drain a target's life force or bestow it to an ally."
+	button_icon = 'yogstation/icons/mob/actions/actions_darkspawn.dmi'
+	button_icon_state = "extract"
+	background_icon_state = "bg_alien"
+	overlay_icon_state = "bg_alien_border"
+	buttontooltipstyle = "alien"
+	panel = "Darkspawn"
+	antimagic_flags = MAGIC_RESISTANCE_MIND
+	check_flags = AB_CHECK_CONSCIOUS | AB_CHECK_HANDS_BLOCKED | AB_CHECK_LYING
+	spell_requirements = SPELL_REQUIRES_HUMAN
+	ranged_mousepointer = 'icons/effects/mouse_pointers/visor_reticule.dmi'
+	cast_range = 5
+	var/mob/living/channeled
+	var/datum/beam/visual
+	var/datum/antagonist/darkspawn/cost
+	var/upkeep_cost = 2 //happens 5 times a second
+	var/damage_amount = 2 //these also happens 5 times a second
+	var/actual_cooldown = 15 SECONDS //this only applies when the channel is broken
+	var/healing = FALSE
+
+/datum/action/cooldown/spell/pointed/extract/New()
+	..()
+	START_PROCESSING(SSfastprocess, src)
```

Comment:
This shouldn't be on when the spell is being granted instead? The process seems to think it will always be in a mob.